src/patterns/ec2-app.test.ts

```diff
@@ -122,24 +136,48 @@ describe("the GuEC2App pattern", function () {
       userData: "",
     });
 
-    expect(stack).toHaveResource("AWS::EC2::SecurityGroup", {
-      SecurityGroupIngress: [
-        {
-          CidrIp: "192.168.1.1/32",
-          Description: "Allow access on port 443 from 192.168.1.1/32",
-          FromPort: 443,
-          IpProtocol: "tcp",
-          ToPort: 443,
-        },
-        {
-          CidrIp: "8.8.8.8/32",
-          Description: "Allow access on port 443 from 8.8.8.8/32",
-          FromPort: 443,
-          IpProtocol: "tcp",
-          ToPort: 443,
-        },
-      ],
-    });
+    const json = SynthUtils.toCloudFormation(stack) as SynthedStack;
+    const securityGroupKey = Object.keys(json.Resources).filter((resource) =>
+      // This will fail if we ever change the security group ID,
+      //if there's a cleaner way of identifying the security group we should do it
+      resource.toLowerCase().startsWith("restrictedingresssecuritygroup")
+    )[0];
+    const securityGroup = json.Resources[securityGroupKey];
+    const loadBalancerKey = Object.keys(json.Resources).filter(
+      (resource) => json.Resources[resource].Type === "AWS::ElasticLoadBalancingV2::LoadBalancer"
+    )[0];
+    const loadBalancer = json.Resources[loadBalancerKey];
+    const lbSecurityGroups = loadBalancer.Properties.SecurityGroups as Array<{
```

Comment:
I wonder if #536 could help here?

Reply:
We've agreed IRL that although this might help, it might not be super straightforward to implement and the current implementation successfully tests what we need it to, so we're going to leave it for now.